crates/shared/src/order_quoting.rs

```diff
@@ -175,9 +177,22 @@ impl Quote {
         }
     }
 
-    /// Applies a subsidy to the quote.
-    pub fn with_subsidy(mut self, subsidy: &Subsidy) -> Self {
-        self.fee_amount = self.data.fee_parameters.subsidized(subsidy);
+    /// Applies a subsidy to the quote **with** the `QuoteSigningScheme.verification_gas_limit`
+    pub fn with_subsidy_and_signing_scheme(
+        mut self,
+        subsidy: &Subsidy,
+        scheme: &QuoteSigningScheme,
+    ) -> Self {
+        let verification_fee = match scheme {
+            QuoteSigningScheme::Eip1271 {
+                verification_gas_limit,
+                ..
+            } => *verification_gas_limit,
+            _ => Default::default(),
+        };
+
+        // THIS CANNOT MODIFY `quote.data`.
+        self.fee_amount = self.data.fee_parameters.subsidized(subsidy) + verification_fee;
```

Comment:
I think that the `verification_fee` should be in gas units (not Eth or traded token). This means that this needs to be passed into the `fee_parameters` so that it is:
1. Converted from gas units to the sell token using the `fee_parameters` sell token price and gas estimate
2. Also included in the factor subsidies that we apply

I think we should have:
```rust
impl FeeParameters {
    pub fn subsidized(&self, subsidy: &Subsidy) -> U256 {
        self.subsidized_with_additional_cost(subsidy, 0)
    }
    
    pub fn subsidized_with_additional_cost(&self, subsidy: &Subsidy, additional_cost: U256) -> U256 {
        let fee_in_eth = (self.gas_amount + additional_cost) * self.gas_price;
        // same as before...
    }
}
```

And then
```suggestion
        // THIS CANNOT MODIFY `quote.data`.
        self.fee_amount = self.data.fee_parameters.subsidized_with_additional_cost(subsidy, verification_fee);
```

Reply:
fixed.